### PR TITLE
feat(web-cloud): multisite - detach a domain

### PR DESCRIPTION
**Pull request by dectotam**
| Question         | Answer
| ---------------- | ---
| Branch?          | master
| Bug fix?         | no
| New feature?     | no
| Breaking change? | no
| Tickets          | Feat MANAGER-12657
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description

Web Cloud - Hosting - Multisite - detach a domain name

## Related

##Related

### Translations
- [x] CMT-6447

### CODE REVIEW

File: packages/manager/apps/web/client/app/hosting/multisite/delete/hosting-multisite-delete.controller.js [hunk 1]
```diff
@@ -46,15 +55,35 @@ angular.module('App').controller(
...
+        this.Domain.getAllNameServer(this.selected.domain.name);
+
+      promises.push(
+        getDomainsService()
+          .then(() => getZoneByZoneName())
+          .then(() => getAllNameServer())
+          .then(() => {
+            this.isOvhDomain = true;
+          }),
+      );
```
Comment (anooparveti) [inline packages/manager/apps/web/client/app/hosting/multisite/delete/hosting-multisite-delete.controller.js:74]: **suggestion:**
```suggestion
      const getDomainsService = () =>
        this.Domain.getDomainsService(this.selected.domain.name);

      promises.push(
        getDomainsService()
          .then(() => this.Domain.getZoneByZoneName(this.selected.domain.name))
          .then(() => this.Domain.getAllNameServer(this.selected.domain.name))
          .then(() => {
            this.isOvhDomain = true;
          }),
      );
```

File: packages/manager/apps/web/client/app/hosting/multisite/delete/hosting-multisite-delete.controller.js [hunk 2]
```diff
@@ -46,15 +55,35 @@ angular.module('App').controller(
...
+        this.Domain.getZoneByZoneName(this.selected.domain.name);
+      const getAllNameServer = () =>
+        this.Domain.getAllNameServer(this.selected.domain.name);
+
+      promises.push(
+        getDomainsService()
+          .then(() => getZoneByZoneName())
+          .then(() => getAllNameServer())
+          .then(() => {
+            this.isOvhDomain = true;
```
Comment (anooparveti) [inline packages/manager/apps/web/client/app/hosting/multisite/delete/hosting-multisite-delete.controller.js:72]: **question:** How can you be sure the domain belongs to OVH? Do we have an error if it's not?

**Review by ghyenne — DISMISSED** [review]

File: packages/manager/apps/web/client/app/hosting/multisite/delete/hosting-multisite-delete.controller.js (hunk 2 above)
Comment (dectotam) [reply packages/manager/apps/web/client/app/hosting/multisite/delete/hosting-multisite-delete.controller.js:72]: isOvhDomain is set to false at first. Then, the three API calls needs to return a 200 OK. If so, the domain is an OVH one, so I set the variable to true in the last successful callback. If one of the calls returns an error, then no need to display anything specific I think, the variable isOvhDomain will then stay to false

File: packages/manager/apps/web/client/app/hosting/multisite/delete/hosting-multisite-delete.controller.js [hunk 3]
```diff
@@ -46,15 +55,35 @@ angular.module('App').controller(
...
+        this.Domain.getDomainsService(this.selected.domain.name);
+      const getZoneByZoneName = () =>
+        this.Domain.getZoneByZoneName(this.selected.domain.name);
+      const getAllNameServer = () =>
+        this.Domain.getAllNameServer(this.selected.domain.name);
+
+      promises.push(
+        getDomainsService()
+          .then(() => getZoneByZoneName())
+          .then(() => getAllNameServer())
```
[the remark this answers is not in the document]
Comment (dectotam) [reply packages/manager/apps/web/client/app/hosting/multisite/delete/hosting-multisite-delete.controller.js:70]: If would imply to add a catch to each one by setting ovhDomain to false if so, isn't it "less pretty" ?

[the remark this answers is not in the document]
Comment (darsene) [reply packages/manager/apps/web/client/app/hosting/multisite/delete/hosting-multisite-delete.controller.js:70]: no, you add your catch after `$q.all`

**Review by darsene — APPROVED** [review]

**Review by anooparveti — DISMISSED** [review]

**Review by antonymarion — DISMISSED** [review]

**Review by ghyenne — DISMISSED** [review]

**Review by darsene — APPROVED** [review]

**Review by darsene — DISMISSED** [review]

**Review by sachinrameshn — DISMISSED** [review]

**Review by ghyenne — DISMISSED** [review]

File: packages/manager/apps/web/client/app/hosting/multisite/delete/hosting-multisite-delete.controller.js [hunk 4]
```diff
@@ -11,13 +12,20 @@ angular.module('App').controller(
...
     ) {
       this.$scope = $scope;
       this.$stateParams = $stateParams;
       this.$translate = $translate;
       this.atInternet = atInternet;
       this.Alerter = Alerter;
       this.HostingDomain = HostingDomain;
+      this.coreURLBuilder = coreURLBuilder;
+      this.Domain = Domain;
+      this.$q = $q;
```
Comment (JacquesLarique) [inline packages/manager/apps/web/client/app/hosting/multisite/delete/hosting-multisite-delete.controller.js:27]: These seem unused

**Review by sachinrameshn — DISMISSED** [review]

**Review by antonymarion — DISMISSED** [review]

**Review by antonymarion — APPROVED** [review]

**Pull request merged**